mirai-core/src/androidMain/kotlin/utils/crypto/RSAAndroid.kt

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2019-2023 Mamoe Technologies and contributors.
+ *
+ * 此源代码的使用受 GNU AFFERO GENERAL PUBLIC LICENSE version 3 许可证的约束, 可以在以下链接找到该许可证.
+ * Use of this source code is governed by the GNU AGPLv3 license that can be found through the following link.
+ *
+ * https://github.com/mamoe/mirai/blob/dev/LICENSE
+ */
+
+package net.mamoe.mirai.internal.utils.crypto
+
+import net.mamoe.mirai.utils.decodeBase64
+import net.mamoe.mirai.utils.encodeBase64
+import java.security.KeyFactory
+import java.security.KeyPairGenerator
+import java.security.SecureRandom
+import java.security.spec.PKCS8EncodedKeySpec
+import java.security.spec.X509EncodedKeySpec
+import javax.crypto.Cipher
+
+internal actual fun rsaEncryptWithX509PubKey(input: ByteArray, pubPemKey: ByteArray, seed: Long): ByteArray {
+    val encodedKey = pubPemKey.decodeToString()
```

Comment:
```suggestion
internal actual fun rsaEncryptWithX509PubKey(input: ByteArray, pubPemKey: String, seed: Long): ByteArray {
    val encodedKey = pubPemKey.decodeToString()
```
你处理的是 String, 那就应该传 String, 但实际上你应该在别的函数处理解码 `-----BEGIN PUBLIC KEY-----` 的过程

Reply:
传 String 正确的，但是 JVM target 的处理还是就在这吧，因为 native target 不用处理